.github/workflows/deploy.yml

```
@@ -8,6 +8,10 @@ on:
     tags:
       - rust-1.**
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

```suggestion
concurrency:
  group: ${{ github.workflow }}
  cancel-in-progress: false
```

I think this would be the correct behavior.

Your fix works for commits pushed to the master branch. But the deploy job also runs on tag push and pushes to the beta branch. On those 2 refs, the `deploy.sh` script does something slightly different. So if a push to the master branch would cancel the workflow on beta/tag, then the commit for those would never be pushed.

The beta/tag push suffers from the same thing, see [book](https://doc.rust-lang.org/clippy/development/infrastructure/release.html#tag-the-stable-commit). 

Deployment will be slower by queuing everything, but not canceling anything. But I don't think this matters, as usually one doesn't wait for this.

With this, you can update the book and remove the note of waiting for the deploy job.

---

Thanks! Made a change and removed the notation in the book.